src/components/application_manager/src/commands/command_request_impl.cc

```diff
@@ -789,75 +808,73 @@ bool CommandRequestImpl::IsResultCodeUnsupported(
+      return "SDL";
+    default:
+      return "Unknown type";
+  }
+}
+
 void CommandRequestImpl::AddTimeOutComponentInfoToMessage(
-    smart_objects::SmartObjectSPtr& response) const {
+    smart_objects::SmartObject& response,
+    const std::set<HmiInterfaces::InterfaceID>& not_responding_interfaces)
+    const {
   using NsSmartDeviceLink::NsSmartObjects::SmartObject;
   LOG4CXX_AUTO_TRACE(logger_);
 
-  if (!response) {
-    LOG4CXX_WARN(logger_, "Invalid message object");
-    return;
-  }
-
-  const uint32_t app_connection_key = connection_key();
-  const ApplicationSharedPtr app =
-      application_manager_.application(app_connection_key);
-  if (!app) {
-    LOG4CXX_WARN(logger_, "Invalid connection_key: " << app_connection_key);
-    return;
-  }
-
-  const SmartObject* app_type = app->app_types();
-  if (!app_type) {
-    LOG4CXX_WARN(logger_, "Empty application types array!");
-    return;
-  }
-
-  if (NsSmartDeviceLink::NsSmartObjects::SmartType_Array !=
-      app_type->getType()) {
-    LOG4CXX_WARN(logger_, "Application types are not an array!");
-    return;
-  }
-
-  const SmartObject& app_type_so = app_type->getElement(0);
-  if (NsSmartDeviceLink::NsSmartObjects::invalid_object_value == app_type_so) {
-    LOG4CXX_WARN(logger_, "Invalid object for component type!");
+  if (not_responding_interfaces.empty()) {
+    LOG4CXX_DEBUG(logger_,
+                  "not_responding_interfaces is empty, nothing to do. return.");
     return;
   }
 
-  const int64_t app_hmi_type = app_type_so.asInt();
-  const std::string& component_name = AppHMITypeToString(
-      static_cast<mobile_apis::AppHMIType::eType>(app_hmi_type));
-  const std::string component_info =
-      component_name + " component does not respond";
-  (*response)[strings::msg_params][strings::info] = component_info;
-}
-
-std::string CommandRequestImpl::AppHMITypeToString(
-    const mobile_apis::AppHMIType::eType app_hmi_type) const {
-  switch (app_hmi_type) {
-    case mobile_apis::AppHMIType::DEFAULT:
-      return "Default";
-    case mobile_apis::AppHMIType::COMMUNICATION:
-      return "Communication";
-    case mobile_apis::AppHMIType::MEDIA:
-      return "Media";
-    case mobile_apis::AppHMIType::MESSAGING:
-      return "Messaging";
-    case mobile_apis::AppHMIType::NAVIGATION:
-      return "Navigation";
-    case mobile_apis::AppHMIType::INFORMATION:
-      return "Information";
-    case mobile_apis::AppHMIType::SOCIAL:
-      return "Social";
-    case mobile_apis::AppHMIType::BACKGROUND_PROCESS:
-      return "Background Process";
-    case mobile_apis::AppHMIType::TESTING:
-      return "Testing";
-    case mobile_apis::AppHMIType::SYSTEM:
-      return "System";
-    default:
-      return "Invalid Component Type";
+  // We copy contents to vector so we can have iterator that will allow us to do
+  // operator-() on it, it will be used in the check later
+  std::vector<HmiInterfaces::InterfaceID> not_responding_interfaces_vector;
```

Comment:
@LuxoftAKutsan about solution with "accumulate": template type of set is enum, not a string, so I still will have to convert set to set of strings or vector of strings for your solution to work.